.github/workflows/buildAndroid.yml

```diff
@@ -0,0 +1,188 @@
+        with:
+          ref: ${{ inputs.ref }}
+
+      - name: Configure MapBox SDK
+        run: ./scripts/setup-mapbox-sdk.sh ${{ secrets.MAPBOX_SDK_DOWNLOAD_TOKEN }}
+
+      - name: Setup Node
+        uses: ./.github/actions/composite/setupNode
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: oracle
+          java-version: 17
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1.190.0
+        with:
+          bundler-cache: true
+
+      - name: Decrypt keystore and json key
+        run: |
+          cd android/app
+          gpg --batch --yes --decrypt --passphrase="${{ secrets.LARGE_SECRET_PASSPHRASE }}" --output my-upload-key.keystore my-upload-key.keystore.gpg
+          gpg --batch --yes --decrypt --passphrase="${{ secrets.LARGE_SECRET_PASSPHRASE }}" --output android-fastlane-json-key.json android-fastlane-json-key.json.gpg
+
+      - name: Get package version
+        id: getPackageVersion
+        run: echo "VERSION=$(jq -r .version < package.json)" >> "$GITHUB_OUTPUT"
+
+      - name: Get Android native version
+        id: getAndroidVersion
+        run: echo "VERSION_CODE=$(grep -o 'versionCode\s\+[0-9]\+' android/app/build.gradle | awk '{ print $2 }')" >> "$GITHUB_OUTPUT"
+
+      - name: Setup DotEnv
+        if: ${{ inputs.type != 'release' }}
+        run: |
+          if [ '${{ inputs.type }}' == 'adhoc' ]; then
+            cp .env.staging .env.adhoc
+            sed -i 's/ENVIRONMENT=staging/ENVIRONMENT=adhoc/' .env.adhoc
+            echo "PULL_REQUEST_NUMBER=${{ inputs.pull_request_number }}" >> .env.adhoc
+          else
+            envFile=''
+            if [ '${{ inputs.type }}' == 'e2e' ]; then
+              envFile='tests/e2e/.env.e2e'
+            else
+              envFile=tests/e2e/.env.e2edelta
+            fi
+            {
+              echo "EXPENSIFY_PARTNER_NAME=${{ secrets.EXPENSIFY_PARTNER_NAME }}"
+              echo "EXPENSIFY_PARTNER_PASSWORD=${{ secrets.EXPENSIFY_PARTNER_PASSWORD }}"
+              echo "EXPENSIFY_PARTNER_USER_ID=${{ secrets.EXPENSIFY_PARTNER_USER_ID }}"
+              echo "EXPENSIFY_PARTNER_USER_SECRET=${{ secrets.EXPENSIFY_PARTNER_USER_SECRET }}"
+              echo "EXPENSIFY_PARTNER_PASSWORD_EMAIL=${{ secrets.EXPENSIFY_PARTNER_PASSWORD_EMAIL }}"
+            } >> "$envFile"
+          fi
+
+      - name: Build Android app
+        id: build
+        run: |
+          lane=''
+          case '${{ inputs.type }}' in
+            'release')
+              lane='build';;
+            'adhoc')
+              lane='build_adhoc';;
+            'e2e')
+              lane='build_e2e';;
+            'e2eDelta')
+              lane='build_e2eDelta';;
+          esac
+          bundle exec fastlane android "$lane"
+
+          # Reload environment variables from GITHUB_ENV
+          # shellcheck disable=SC1090
+          source "$GITHUB_ENV"
```

Comment:
What does this do and why is it required?

Reply:
Explained in [this comment](https://github.com/Expensify/App/pull/49434#discussion_r1782029145), but let me add a permanent code comment

Reply:
Tried to improve the code comment, let me know what you think